### DISCUSSION TITLE
How To Use KEDA external-scaler-azure-cosmos-db in Azure Function App Under AKS

### DISCUSSION THREAD

**Discussion by SOFSPEEL** [kedacore/external-scaler-azure-cosmos-db#57] [General]
When running "traditional" Azure Function Apps (ie. not under AKS no KEDA) needing a CosmosDBTrigger one creates the CosmosDBTrigger and a feed processor does **not** need to be specified, my understanding is that the feed processor in this scenario is handled automagically for you. This is a nice feature of Azure Function Apps.

So, to use external-scaler-azure-cosmos-db under AKS in a function app, it is mandatory to create a feed processor: https://github.com/kedacore/external-scaler-azure-cosmos-db#:~:text=the%20change%20feeds.-,processorName,-%2D%20Name%20of%20change

So two points/questions:
- This implies that the external-scaler-azure-cosmos-db is not very convenient to use, i.e in non-AKS scenario there was no need to create the feed processor.
- In the context of an Azure Function app running under AKS, how would one even create a feed processor, as Azure functions are intended to be discrete, short-running functions, not long-running hosts as required to host a feed processor?

General question: can external-scaler-azure-cosmos-db even be used in Azure Function apps under AKS?

**Comment by zroubalik** [comment]
@tomkerkhove FYI

**Reply by tomkerkhove** [reply]
Adding @raorugan who should be able to help

**Comment by raorugan** [comment]
can you pls check out this sample deployment for AKS - https://github.com/kedacore/external-scaler-azure-cosmos-db/blob/main/src/Scaler.Demo/README.md

**Reply by SOFSPEEL** [reply]
@raorugan so the sample shows how to implement this as a long-running process: see https://github.com/kedacore/external-scaler-azure-cosmos-db/blob/main/src/Scaler.Demo/OrderProcessor/Dockerfile. It doesn't show how this would work as a Function App (i.e. not long-running process).
